mpmp/prediction/cross_validation.py

```diff
@@ -337,6 +365,191 @@ def run_cv_stratified(data_model,
     return results
 
 
+def run_cv_fold(data_model,
+                exp_string,
+                identifier,
+                training_data,
+                sample_info,
+                num_folds,
+                fold_no,
+                shuffle_labels=False,
+                standardize_columns=False,
+                num_features=-1,
+                feature_selection_method='mad',
+                bayes_opt=False,
+                nonlinear=False,
+                output_preds=False,
+                output_grid=False):
+
+    results = {
+        '{}_metrics'.format(exp_string): [],
+        '{}_auc'.format(exp_string): [],
+        '{}_aupr'.format(exp_string): [],
+    }
+    signal = 'shuffled' if shuffle_labels else 'signal'
+
+    if output_preds:
+        results['{}_preds'.format(exp_string)] = []
+
+    if output_grid:
+        results['{}_param_grid'.format(exp_string)] = []
+
+    try:
+        with warnings.catch_warnings():
+            # sklearn warns us if one of the stratification classes has fewer
+            # members than num_folds: in our case that will be the 'other'
+            # class, and it's fine to distribute those unevenly. so here we
+            # can ignore that warning.
+            warnings.filterwarnings('ignore',
+                                    message='The least populated class in y')
+            X_train_raw_df, X_test_raw_df, _ = split_stratified(
+               data_model.X_df, sample_info, num_folds=num_folds,
+               fold_no=fold_no, seed=data_model.seed)
+    except ValueError:
+        if data_model.X_df.shape[0] == 0:
+            raise NoTrainSamplesError(
+                'No train samples found for identifier: {}'.format(
+                    identifier)
+            )
+
+    y_train_df = data_model.y_df.reindex(X_train_raw_df.index)
+    y_test_df = data_model.y_df.reindex(X_test_raw_df.index)
+
+    # shuffle labels for train/test sets separately
+    # this ensures that overall label balance isn't affected
+    # (see https://github.com/greenelab/mpmp/issues/44)
+    if shuffle_labels:
+        # in this case we want to shuffle labels independently for each cancer type
+        # (i.e. preserve the total number of mutated samples in each)
+        original_ones = y_train_df.groupby('DISEASE').sum()['status']
+        y_train_df.status = shuffle_by_cancer_type(y_train_df, data_model.seed)
+        y_test_df.status = shuffle_by_cancer_type(y_test_df, data_model.seed)
+        new_ones = y_train_df.groupby('DISEASE').sum()['status']
+
+        # number of mutated samples per cancer type should be the same before
+        # and after shuffling
+        assert original_ones.equals(new_ones)
+
```

Comment:
Triple newline should be single newline for PEP reasons (sorry I can't figure out how to remove multiple newlines in a single suggestion).
```suggestion
```